crates/ra_ide/src/completion/complete_keyword.rs

```diff
@@ -121,13 +111,62 @@ fn complete_return(
 
 #[cfg(test)]
 mod tests {
-    use crate::completion::{test_utils::do_completion, CompletionItem, CompletionKind};
+    use crate::{
+        completion::{
+            test_utils::{do_completion, do_completion_with_position},
+            CompletionItem, CompletionKind,
+        },
+        CompletionItemKind,
+    };
     use insta::assert_debug_snapshot;
 
     fn do_keyword_completion(code: &str) -> Vec<CompletionItem> {
         do_completion(code, CompletionKind::Keyword)
     }
 
+    fn get_completion_text_and_assert_positions(code: &str) -> Vec<(String, String)> {
+        let (position, completion_items) =
+            do_completion_with_position(code, CompletionKind::Keyword);
+        let mut returned_keywords = Vec::<(String, String)>::new();
+
+        for item in completion_items {
+            debug_assert!(item.text_edit().len() == 1);
+            assert!(item.kind() == Some(CompletionItemKind::Keyword));
+            let atom = item.text_edit().iter().next().unwrap().clone();
+            assert!(atom.delete.start() == position.offset);
+            assert!(atom.delete.end() == position.offset);
+            let pair = (item.label().to_string(), atom.insert);
+            returned_keywords.push(pair);
+        }
+        returned_keywords.sort();
+        returned_keywords
+    }
+
+    #[test]
+    fn completes_keywords_in_use_stmt_new_approach() {
+        assert_debug_snapshot!(
```

Comment:
Just FYI, here is the last version which on the one hand reduces the amount of data needed for snapshot, and on the other hand still relies on them for diffs